modules/local/idSnp/main.nf

```diff
@@ -0,0 +1,153 @@
+        """
+}
+
+process SNP_CHECK {
+    label 'process_single'
+    tag "${meta.id}"
+
+    input:
+        tuple val(group), val(meta), file(vcfs)
+
+    output:
+        tuple val(group), file("*.csv"),            emit: idsnp_checked
+        path "versions.yml", optional: true         ,emit:   versions
+
+    when:
+        task.ext.when == null || task.ext.when
+
+    script:
+        tumor_idx   = meta.type.findIndexOf{ it == 'tumor' || it == 'T' }
+        normal_idx  = meta.type.findIndexOf{ it == 'normal' || it == 'N' }
+        normal_id   = meta.id[normal_idx]
+        tumor_id    = meta.id[tumor_idx]
+	    normalvcf   = vcfs[normal_idx]
+	    tumorvcf    = vcfs[tumor_idx]
+
+        if(meta.id.size() == 2) {
+            """
+            idsnp_controller-myeolid_specific.pl \\
+                --vcf_sample $tumorvcf  \\
+                --vcf_control $normalvcf \\
+                --sample  $tumor_id \\
+                --control $normal_id 
+        
+            cat <<-END_VERSIONS > versions.yml
+            "${task.process}":
+                perl: \$( echo \$(perl -v 2>&1) |sed 's/.*(v//; s/).*//')
+            END_VERSIONS
+            """
+        } else {
+            """
+            echo "Not applicable" > s${tumor_id}.csv
+            
+            cat <<-END_VERSIONS > versions.yml
+            "${task.process}":
+                perl: \$( echo \$(perl -v 2>&1) |sed 's/.*(v//; s/).*//')
+            END_VERSIONS
+            """
+        }
+
+    stub:
+        tumor_idx   = meta.type.findIndexOf{ it == 'tumor' || it == 'T' }
+        normal_idx  = meta.type.findIndexOf{ it == 'normal' || it == 'N' }
+        normal_id   = meta.id[normal_idx]
+        tumor_id    = meta.id[tumor_idx]
+        
+        if(meta.id.size() == 2) {
+            """ 
+            touch s${tumor_id}_c${normal_id}.csv
+
+            cat <<-END_VERSIONS > versions.yml
+            "${task.process}":
+                perl: \$( echo \$(perl -v 2>&1) |sed 's/.*(v//; s/).*//')
+            END_VERSIONS
+            """
+        } else {
+        
+            """
+            echo "Not applicable" > s${tumor_id}.csv
+
+            cat <<-END_VERSIONS > versions.yml
+            "${task.process}":
+                perl: \$( echo \$(perl -v 2>&1) |sed 's/.*(v//; s/).*//')
+            END_VERSIONS
+            """
+        }
+}
+
+process PROVIDER {
+    label 'process_single'
+    tag "${meta.id}"
+
+    input:
+        tuple val(group), val(meta), file(bam), file(bai)
+
+    
+    output:
+        tuple val(group), val(meta), file("*.genotypes"),    emit: genotype_checked
+        path "versions.yml",                                 emit: versions  // Emit version information in YAML format
+    
+    when:
+        task.ext.when == null || task.ext.when
+        
+    script:
+        def prefix  = task.ext.prefix ?: "${meta.id}"
+        def args    = task.ext.args  ?: ""
+
+        """
+        provider.pl  --bam $bam  $args  --out $prefix 
```

Comment:
Will the script automatically append .genotypes to the output file?

Reply:
Yes , the script [provider.pl](https://github.com/Clinical-Genomics-Lund/SomaticPanelPipeline/blob/features/idsnpQC/bin/provider.pl) has a subroutine called print_genotype_table that emits genotype tables.